app/Http/Controllers/Order/ExtendedOrderController.php

```diff
@@ -116,6 +118,62 @@ public function allInstallations($allInstallation,$join)
         return $join;
     }
 
+    /**
+     * Searches for InActive Installation
+     *
+     * @param  string $allInstallation
+     * @param  App\Model\Order $join The order instance
+     *
+     * @return $join
+     */
+    public function allInactiveInstallations($allInstallation,$join)
+    {
+        if($allInstallation) {
+            $dayUtc = new Carbon('-30 days');
+            $minus30Day = $dayUtc->toDateTimeString();
+            if($allInstallation == 'paid_inactive_ins') {
+                $join = $join->where('price_override', '>', 0)->whereHas('subscription', function($query) use($minus30Day) {
+                    $query->where('updated_at', '<', $minus30Day);
+                });
+            } elseif($allInstallation == 'unpaid_inactive_ins') {
+                $join = $join->where('price_override', '=', 0)->whereHas('subscription', function($query) use($minus30Day) {
+                    $query->where('updated_at', '<', $minus30Day);
+                });
+            } elseif ($allInstallation == 'all_inactive_ins') {
+                $join = $join->whereHas('subscription', function($query) use($minus30Day) {
+                    $query->where('updated_at', '<', $minus30Day);
+                });
+            }
+        }
+        return $join;
+    }
+
+    /**
+     * Searches for Renewals
+     *
+     * @param  string $allInstallation
+     * @param  App\Model\Order $join The order instance
+     *
+     * @return $join
+     */
+    protected function allRenewals($allRenewal,$join)
+    {
+        if($allRenewal) {
+            $dayUtc = new Carbon();
+            $now = $dayUtc->toDateTimeString();
+            if($allRenewal == 'expired_subscription') {
+                $join = $join->whereHas('subscription', function($query) use($now){
+                    $query->where('update_ends_at', '<', $now);
```

Comment:
you can use `$allRenewal ` inside `use` and add this condition inside subquery. Something like 
```
return $join->whereHas('subscription', function($query) use($allRenewal){
    if($allRenewal == 'expired_subscription'){
    	return $query->where('update_ends_at', '>', $now);
    }
    return    $query->where('update_ends_at', '>', $now);
});
```

Reply:
Fixed